client_test/function_serialization_test.py

```diff
@@ -0,0 +1,42 @@
+# Copyright Modal Labs 2023
+import pytest
+
+from modal import Stub
+from modal.app import App
+from modal._serialization import deserialize
+
+
+@pytest.mark.asyncio
+async def test_serialize_deserialize_function(servicer, client):
+    stub = Stub()
+
+    @stub.function(serialized=True, name="foo")
+    def foo():
+        2 * bar.call()
+
+    @stub.function(serialized=True, name="bar")
+    def bar():
+        return 5
+
+    assert foo.object_id is None
+    assert bar.object_id is None
+
+    with stub.run(client=client) as app:
+        pass
+
+    assert foo.object_id is not None
+    assert bar.object_id is not None
+    assert {foo.object_id, bar.object_id} == servicer.reserved_functions
+
+    foo_def = servicer.app_functions[foo.object_id]
+
+    assert len(servicer.client_calls) == 0
+    from modal_utils.async_utils import synchronizer
+
+    container_app = App.init_container(client, app.app_id)
```

Comment:
This is a kind of hacky way to test the deserialization logic, not sure how to best client test this without adding significant additional functionality to the mock servicer and actually running the container_entrypoint on the definition from here... We do have an integration test on the internal repo that tests this properly though